src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/SecurityFactoryInterface.php

```diff
@@ -21,6 +21,21 @@
  */
 interface SecurityFactoryInterface
 {
+    /**
+     * Adds the security listener and authentication provider
+     * to the DI container for the appropriate security context.
+     *
+     * @param ContainerBuilder $container
+     * @param string           $id                The unique id of the firewall 
+     * @param array            $config            The options array for the listener
+     * @param string           $userProvider      The id of the user provider
```

Comment:
The "id" on line 29 references a unique name, while the id here means the service id. Let's make that difference more clear by using "The service id of the user provider" here.